pkg/scheduler/api/types.go

```diff
@@ -29,6 +29,10 @@ const (
 	// and only if, N.IdleResource < T.RequestResource <= N.AllocatableResource.
 	AllocatedOverBackfill
 
+	// OverOccupied means the job is allocated even when there is no immedicately
+	// available resources due to starvation
+	OverOccupied
```

Comment:
Do we need this state? How about we leave the job in `Allocated` state?

Reply:
this state is to keep the task from being dispatched

Reply:
how about merge `AllocatedOverBackfill` and `OverOcupied` into one `Waiting` state?

Reply:
looking into it

Reply:
doesn't look like they can be merged. AllocatedOverBackfill is also used for patching annotation. OverOcupied specifically means this task is allocated using without being able to be dispatched because resource is over-allocated

Reply:
It looks to me that we use `task.IsBackfill` to decide whether or not to patch a pod. We do not use `AllocatedOverBackfill`.